src/pattern-library/components/patterns/prototype/SelectNextPage.tsx

```diff
@@ -0,0 +1,344 @@
+import classnames from 'classnames';
+import { useCallback, useMemo, useState } from 'preact/hooks';
+
+import { ArrowLeftIcon, ArrowRightIcon } from '../../../../components/icons';
+import { IconButton, InputGroup } from '../../../../components/input';
+import Select from '../../../../components/input/SelectNext';
+import Library from '../../Library';
+
+const items = [
+  { id: '1', name: 'All students' },
+  { id: '2', name: 'Albert Banana' },
+  { id: '3', name: 'Bernard California' },
+  { id: '4', name: 'Cecelia Davenport' },
+  { id: '5', name: 'Doris Evanescence' },
+];
+
+function Select_({
+  disabled,
+  textOnly,
+  theItems = items,
+}: {
+  disabled?: boolean;
+  textOnly?: boolean;
+  theItems?: typeof items;
+}) {
+  const [value, setValue] = useState<(typeof theItems)[number]>();
+
+  return (
+    <Select
+      value={value}
+      onChange={setValue}
+      label={
+        value ? (
+          <>
+            {value.name}
+            {!textOnly && (
+              <div className="rounded px-2 bg-grey-7 text-white">
+                {value.id}
+              </div>
+            )}
+          </>
+        ) : disabled ? (
+          <>This is disabled</>
+        ) : (
+          <>Select one...</>
+        )
+      }
+      disabled={disabled}
+    >
+      {theItems.map(item => (
+        <Select.Option value={item} key={item.id}>
+          {() =>
+            textOnly ? (
+              <>{item.name}</>
+            ) : (
+              <>
+                {item.name}
+                <div className="grow" />
+                <div className="rounded px-2 bg-grey-7 text-white">
+                  {item.id}
+                </div>
+              </>
+            )
+          }
+        </Select.Option>
+      ))}
+    </Select>
+  );
+}
+
+function InputGroupSelect_() {
+  const [selected, setSelected] = useState<(typeof items)[number]>();
+  const selectedIndex = useMemo(
+    () => (!selected ? -1 : items.findIndex(item => item === selected)),
+    [selected],
+  );
+  const next = useCallback(() => {
+    const newIndex = selectedIndex + 1;
+    setSelected(
+      (items[newIndex]?.id === '4' ? items[newIndex + 1] : items[newIndex]) ??
```

Comment:
This test is because item 4 is disabled, right? That wasn't immediately obvious. Do we have a use case for disabling specific students in the list currently? If not I'd simplify the example by taking this out.

Reply:
Yep, it's a bit hacky as it takes for granted things happening out of this. I'll simplify it.